pkg/sink/sink_test.go

```diff
@@ -365,20 +364,35 @@ func TestHandleEvent(t *testing.T) {
 	if err != nil {
 		t.Fatalf("Error creating Post request: %s", err)
 	}
+
 	if resp.StatusCode != http.StatusCreated {
 		t.Fatalf("Response code doesn't match: %v", resp.Status)
 	}
 	body, err := ioutil.ReadAll(resp.Body)
 	if err != nil {
 		t.Fatalf("Error reading response body: %s", err)
-		return
 	}
-	if !strings.Contains(string(body), "event (EventID: ") {
-		t.Errorf("Response body doesn't have eventid")
+
+	respData := struct {
+		EventListener string
+		Namespace     string
+		EventID       string
+	}{}
+
+	err = json.Unmarshal(body, &respData)
+	if err != nil {
+		t.Fatalf("Error unmarshalling response body: %s", err)
+	}
+
+	if respData.EventID == "" {
+		t.Fatalf("EventID isn't available: %s", err)
+	}
+	if respData.EventListener != el.Name {
+		t.Errorf("Diff EventListener name: %s, should: %s", respData.EventListener, el.Name)
 	}
-	wantPayload := `EventListener: my-eventlistener in Namespace: foo handling event`
-	if !strings.Contains(string(body), wantPayload) {
-		t.Errorf("Diff response body: %s, should have: %s", string(body), wantPayload)
+
+	if respData.Namespace != el.Namespace {
```

Comment:
optional: Since UID can now be set deterministicly in tests, you could replace this with a diff call instead of doing ifs on each field.